ehsm_kms_service/test/test_kms_with_thread.py

```diff
@@ -0,0 +1,104 @@
+import requests
+import json
+import argparse
+import base64
+import _thread
+import time
+import datetime
+import random
+import hmac
+from hashlib import sha256
+from collections import OrderedDict
+import urllib.parse
+appid= '9524787e-5a35-47a7-b07f-9e1df7afedc4'
+apikey= 'yJvJsS7JZ1mQq8UBaw4Dxy1p1L7Bkvk2'
+keyid= ''
+
+def test_params(payload):
+    params = OrderedDict()
+    params["appid"] = appid
+    if payload!=False:
+        ord_payload = OrderedDict(sorted(payload.items(), key=lambda k: k[0]))
+        params["payload"] = urllib.parse.unquote(urllib.parse.urlencode(ord_payload))
+    params["timestamp"] = str(int(time.time() * 1000))
```

Comment:
add a random to prevent replay:
import os
params["timestamp"] = str(int(time.time() * 1000) + int.from_bytes(os.urandom(2), byteorder='little'))

Reply:
ok, It will be change.